### DISCUSSION TITLE
Feature: Dashboard Section/Card Blueprints

### DISCUSSION THREAD

**Discussion by potelux** [home-assistant/frontend#22087] [Dashboard]
# **Use Case:** 
In the UI, I would like to be able to create a section or card derived from other existing cards.  Essentially, this would move the power user feature of a "Dashboard Strategy" directly to the UI allowing less technical users to create **and share** more complicated cards. Think blueprints for the dashboard. 

### **Example of the issue:**  
If I create a dashboard with several similar, but slightly different pages, I must copy the entire configuration between pages.  If I make a change in one location, I must then make that same change manually in every location I use the similar component.

# **Suggested Solution:** 
Mirror the features blueprints provide to automations.  Create the ability to configure a card outside of a dashboard, provide it a custom-card name, and use it in other dashboards.  

### **Further Explanation:** 
I may frequently place two to three shortcut/scene buttons underneath a larger button that controls the lights in a room.  For example, I have uplights on my trees that are group together so that I can control the brightness and color from a single card.  However, on holidays I use scenes to have the lights be a variety of different colors based on the respective holiday.   I usually place those scene buttons directly below the button that controls the light group.

### **Benefit:** 
By grouping those cards into a reusable card that has templated input values (exactly like a blueprint), I could reuse that component everywhere I want to add scene controls below a light control.  **This would allow much more complicated and _shareable_ UIs without requiring people to install custom components or custom strategies.** It would also reduce the headache for users when they build out a dashboard using custom cards that one day stop being supported.  Pretty much every other benefit that blueprints provides applies here as well.

**EDIT** I just saw [this custom component](https://github.com/custom-cards/decluttering-card) that performs a similar function.  It should serve as an example of what the feature should look like.

**Comment by potelux** [comment]
One major thing I want to point out is that a very common challenge for users (even technical users) when creating and sharing dashboards is that great looking dashboards end up having dozens upon dozens of cards and each one has details specific to the person who made the dashboard.  In other words, without a custom dashboard strategy like [the Mushroom Strategy](https://github.com/AalianKhan/mushroom-strategy), or custom integrations like [Dwain's Dashboard](https://dwainscheeren.github.io/dwains-lovelace-dashboard/v3/index.html) or [UI Lovelace Minimalist](https://ui-lovelace-minimalist.github.io/UI/), users can't share their dashboards. Instead, other users' dashboards just act as an inspiration as you build out your own. Even in those examples where the UI is reusable, there still ends up being a need to use the YAML directly or even spin up a VSCode editor.   

Even Lewis from Everything Smart Home calls it ["hard mode"](https://www.youtube.com/watch?v=A0fMt8IRKoI&t=33s)!

Or there's this discussion on reddit where someone writes, ["strategies are cool, but the barrier of entry is too high for me to make my own"](https://www.reddit.com/r/homeassistant/comments/17hli8b/comment/k6pbjkw/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button).

As Paulus says, it is ["always great to see people create and share strategies"](https://community.home-assistant.io/t/auto-generating-mushroom-dashboard-using-strategies/553429/2).  

This feature would have the potential to create a whole new way of sharing and exploring dashboards, much like the existing [Blueprint Exchange](https://community.home-assistant.io/c/blueprints-exchange).

**Comment by BBE-FR** [comment]
Happy user of the decluttering card that was proposed as an exemple above, I wanted to highlight that there is another/new version of it: the streamline card: 
https://github.com/brunosabot/streamline-card

This one allows some interesting strategies, and is UI compatible in the dashboard.

I think it is an interesting way to propagate modification to all cards.
